### PR TITLE
Policy to require public types for API

### PR DESCRIPTION
**Pull request by guptapriya**
We want to require all new public APIs to have publicly documented arguments and return values.

### CODE REVIEW

File: governance/api-reviews.md [hunk 1]
```diff
@@ -159,6 +159,10 @@ latter prevents the implementation from distinguishing between the caller not
 setting the argument vs. the caller setting the argument to the default value,
 which may be needed when the default behavior is changing.
 
+#### Documented types
+
+Arguments and return values to public APIs must be either be of public types, or inherit from a public type. This ensures that the arguments and return value types are documented and gives users clearer guidance on what can be passed to a public API, and what can they do with the returned values. If it is not desirable for the user to construct these types on their own, one can choose to expose superclass with no constructor, but adequate docstrings.
```
Comment (jsimsa) [inline governance/api-reviews.md:164]: "either be" => "either"

Comment (jsimsa) [reply governance/api-reviews.md:164]: "the arguments and return value types" => "the argument types and return value types"

Comment (jsimsa) [reply governance/api-reviews.md:164]: "clearer" => "clear"

Comment (guptapriya) [reply governance/api-reviews.md]: addressed, thanks!

**Comment by guptapriya** [comment]
Thanks @alextp @jsimsa.
Will wait for @martinwicke to be back to merge this.

**Pull request merged**